pkg/operator/ceph/disruption/machinedisruption/reconcile.go

```diff
@@ -0,0 +1,167 @@
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+
+// ReconcileMachineDisruption reconciles MachineDisruption
+type ReconcileMachineDisruption struct {
+	scheme  *runtime.Scheme
+	client  client.Client
+	log     *capnslog.PackageLogger
+	context *controllerconfig.Context
+}
+
+const (
+	defaultMaxUnavailable   = 1
+	zeroMaxUnavailable      = 0
+	requeueAfterFiveMinutes = 5 * time.Minute
+	requeueAfterOneMinute   = time.Minute
+)
+
+// Reconcile is the implementation of reconcile function for ReconcileMachineDisruption
+// which ensures that the machineDisruptionBudget for the rook ceph cluster is in correct state
+func (r *ReconcileMachineDisruption) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.log.Infof("reconciling %s", request.NamespacedName)
+
+	// Fetching the cephCluster
+	cephClusterInstance := &cephv1.CephCluster{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephClusterInstance)
+	if err != nil {
+		r.log.Errorf("%+v", err)
+		return reconcile.Result{}, err
+	}
+
+	// skipping the reconcile since the feature is switched off
+	if !cephClusterInstance.Spec.DisruptionManagement.ManageMachineDisruptionBudgets {
+		r.log.Infof("Skipping reconcile for cephCluster %s as manageMachineDisruption is turned off", request.Name)
+		return reconcile.Result{}, nil
+	}
+
+	mdb := &healthchecking.MachineDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateMDBInstanceName(request.Name, request.Namespace),
+			Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+		},
+	}
+
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: mdb.GetName(), Namespace: mdb.GetNamespace()}, mdb)
+	if errors.IsNotFound(err) {
+		// If the MDB is not found creating the MDB for the cephCluster
+		maxUnavailable := int32(defaultMaxUnavailable)
+		// Generating the MDB instance for the cephCluster
+		newMDB := &healthchecking.MachineDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateMDBInstanceName(request.Name, request.Namespace),
+				Namespace: cephClusterInstance.Spec.DisruptionManagement.MachineDisruptionBudgetNamespace,
+				Labels: map[string]string{
+					MDBCephClusterNamespaceLabelKey: request.Namespace,
+					MDBCephClusterNameLabelKey:      request.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{cephCluster.ClusterOwnerRef(cephClusterInstance.GetName(), string(cephClusterInstance.GetUID()))},
+			},
+			Spec: healthchecking.MachineDisruptionBudgetSpec{
+				MaxUnavailable: &maxUnavailable,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						MDBCephClusterNameLabelKey:      request.Name,
+						MDBCephClusterNamespaceLabelKey: request.Namespace,
+					},
+				},
+			},
+		}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: newMDB.GetName(), Namespace: newMDB.GetNamespace()}, newMDB)
```

Comment:
Why are we doing a GET when we are inside the `if errors.IsNotFound(err)` block of an identical GET? (In a GET only the name and namespace should matter)

Reply:
yeah its wrong. I forgot to remove this after the last refactoring.